api/v1/swagger/definitions/erm_counter_file.yaml

```diff
@@ -0,0 +1,43 @@
+---
+type: object
+properties:
+  erm_counter_files_id:
+    type: integer
+    description: internally assigned counter_files identifier
+    readOnly: true
+  harvester_id:
+    description: harvester_id of the counter_files
+    type:
+    - integer
+    - "null"
```

Comment:
This "null" is setting these fields as nullable but is this correct? They are marked as "required" down below.

Reply:
This is where our specs are a bit annoying.. and partly limited by being in swagger 2 still.

The `required` means the field is required to exist in the data sent/returned. Being `nullable` is more about what said field then contains.

There's also `readOnly`, but that adds even more complication.

The challenge we have, is we're sharing schema for input and output and strictly we don't support PATCH in our API's.. so for PUT and POST all the fields should be `required` as strictly those verbs mean 'full update' and should require the whole object data passed each time (This isn't strictly the case in our API's however.. we're not very consistent here).